python_modules/dagster-graphql/dagster_graphql/schema/instigation.py

```diff
@@ -638,25 +641,54 @@ def resolve_repositoryLocationName(self, _graphene_info: ResolveInfo):
 
     def resolve_hasStartPermission(self, graphene_info: ResolveInfo):
         if self._instigator_state.instigator_type == InstigatorType.SCHEDULE:
-            return graphene_info.context.has_permission_for_location(
-                Permissions.START_SCHEDULE, self._instigator_state.repository_selector.location_name
-            )
+            if self._batch_loader:
+                if self._batch_loader.repository.has_schedule(self._instigator_state.name):
```

Comment:
maybe worth a comment here explaining that the motivation is to use the repository if we already have it and it is being used to batch load a bunch of definitions vs. potentially doing a point query on a large number of selectors